atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepableCells.java

```diff
@@ -131,20 +144,33 @@ SweepBatch getBatchForPartition(ShardAndStrategy shardStrategy, long partitionFi
         SweepableCellsTable.SweepableCellsRow row = computeRow(partitionFine, shardStrategy);
```

Comment:
`getBatchForPartition` now does the following:
1. Fetches the values from the `SweepableCellsTable`
2. Groups the writes by start timestamp, filtering out those that were not committed
3. Bundles the writes that were committed into a `SweepBatch`

This seems like too many things for one method, and I just took ten minutes trying to understand it all. However, extracting methods might be tricky, because we compute the last swept ts along the way. Might be worth doing though, if it's not too gnarly (don't introduce `Pair<lastSweepTs, writes>` for example).

If it would not be straightforward or nice to cut this method down to size, then adding a few comments would make it more readable.

Reply:
Refactored

Reply:
Nice! 🎉